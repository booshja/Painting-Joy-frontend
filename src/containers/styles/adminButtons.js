// dependencies
import styled from "styled-components";

export const StyledGreenSoloButton = styled.button`
    background-color: #207070;
    font-family: "News Cycle", sans-serif;
    font-weight: 700;
    font-size: 1rem;
    letter-spacing: 1px;
    padding: 5px 10px;
    border: none;
    border-radius: 4px;
    color: #ffffff;
`;

export const StyledOutlineButton = styled.button`
    border-radius: 4px;
    background-color: #ffffff;
    font-family: "News Cycle", sans-serif;
    font-weight: 700;
    font-size: 1.2rem;
    letter-spacing: 1px;
    padding: 2.5px 5px;
    border: 2px solid ${(props) => props.color};
    color: ${(props) => props.color};
`;
