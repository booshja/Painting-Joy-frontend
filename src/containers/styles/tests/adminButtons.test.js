import { render } from "@testing-library/react";
import { StyledGreenSoloButton, StyledOutlineButton } from "../adminButtons";

/** Smoke Test */
it("renders without crashing, StyledGreenSoloButton", () => {
    render(<StyledGreenSoloButton>Apple</StyledGreenSoloButton>);
});

it("renders without crashing, StyledOutlineButton, no color prop", () => {
    render(<StyledOutlineButton>Banana</StyledOutlineButton>);
});

it("renders without crashing, StyledOutlineButton, with color prop", () => {
    render(<StyledOutlineButton color="#ffff00">Banana</StyledOutlineButton>);
});

/** Snapshot Test */
it("matches snapshot, StyledGreenSoloButton", () => {
    const { asFragment } = render(
        <StyledGreenSoloButton>Apple</StyledGreenSoloButton>
    );
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot, StyledOutlineButton, no color prop", () => {
    const { asFragment } = render(
        <StyledOutlineButton>Banana</StyledOutlineButton>
    );
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot, StyledOutlineButton, with color prop", () => {
    const { asFragment } = render(
        <StyledOutlineButton color="#ffff00 ">Banana</StyledOutlineButton>
    );
    expect(asFragment()).toMatchSnapshot();
});

/** RTL Tests */
it("renders correctly, StyledGreenSoloButton", () => {
    const { getByText } = render(
        <StyledGreenSoloButton>Apple</StyledGreenSoloButton>
    );

    getByText("Apple");
});

it("renders correctly, StyledOutlineButton, no color prop", () => {
    const { getByText } = render(
        <StyledOutlineButton>Banana</StyledOutlineButton>
    );

    getByText("Banana");
});

it("renders correctly, StyledOutlineButton, with color prop", () => {
    const { getByText } = render(
        <StyledOutlineButton color="#ffff00">Banana</StyledOutlineButton>
    );

    getByText("Banana");
});
