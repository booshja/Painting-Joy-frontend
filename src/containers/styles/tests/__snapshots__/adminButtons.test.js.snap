// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot, StyledGreenSoloButton 1`] = `
<DocumentFragment>
  <button
    class="sc-bdnxRM gcBejj"
  >
    Apple
  </button>
</DocumentFragment>
`;

exports[`matches snapshot, StyledOutlineButton, no color prop 1`] = `
<DocumentFragment>
  <button
    class="sc-gtsrHT larCzd"
  >
    Banana
  </button>
</DocumentFragment>
`;

exports[`matches snapshot, StyledOutlineButton, with color prop 1`] = `
<DocumentFragment>
  <button
    class="sc-gtsrHT ckigvW"
    color="#ffff00 "
  >
    Banana
  </button>
</DocumentFragment>
`;
